app/controllers/events/volunteerRegisterEvent.py

```diff
@@ -0,0 +1,34 @@
+from app.models import*
+from app.models.event import Event
+from app.models.eventParticipant import EventParticipant
+from app.models.user import User
+from app.controllers.events.programEligibility import isEligibleForProgram
+import pytest
+
+def volunteerRegister(userid,  eventid):
+    """
+    Lets the user register for an event if they are eligible
+    :param userid: indicates the id that the user has
+    :param eventid: indicates the id that the event has
+    :return: a message that indicates what the user has registered for if eligible;
+    otherwise lets the user know that they are not eligible for registration
+    """
+
+    user = User.get(User.username == userid)
+    event = Event.get(Event.id == eventid)
+    program = Event.select(Event.program).where(Event.id == eventid)
+    # Assuming the student meets the requirement for the events (function wriiten
+    # by Zach and KArina)
+
+    if isEligibleForProgram(program, user):
+        #checks if user isn't already registerd for event
+        if not EventParticipant.select().where(EventParticipant.user == user,
+                                           EventParticipant.event == event).exists():
+            eventParticipant = EventParticipant.create(user = user,
+                                                       event = event,
+                                                       rsvp = True)
+        else:
+            return f"{user} is already registered for {event}"
+        return eventParticipant
+    else:
+        return ("User is not eligible for the program")
```

Comment:
The function shouldn't return different data types depending on the status of the function. That is, success and failure should be clear from running the function or the return values.

In this case it makes sense to raise an exception for the failure case, and ignore if the user is already registered and return the same value as if the record was created.

Reply:
Refactored the function by raising an exception and returning the value for the created record